### ISSUE TITLE
Package is not installable with latest Python 3.11

### ISSUE THREAD

**Issue by kchoudhu** [aws/aws-cli#7767]
### Describe the bug

Stale pinned versions of PyYAML mean that awscli cannot be installed for the latest Python release (3.11)

### Expected Behavior

awscli is installed

### Current Behavior

Error attempting to install from pip

### Reproduction Steps

```> python3.11 -m pip install awscli --user --ignore-installed

Collecting awscli
  Downloading awscli-1.27.95-py3-none-any.whl (4.0 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.0/4.0 MB 5.6 MB/s eta 0:00:00
Collecting s3transfer<0.7.0,>=0.6.0
  Downloading s3transfer-0.6.0-py3-none-any.whl (79 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 79.6/79.6 KB 16.8 MB/s eta 0:00:00
Collecting botocore==1.29.95
  Downloading botocore-1.29.95-py3-none-any.whl (10.5 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 10.5/10.5 MB 27.0 MB/s eta 0:00:00
Collecting colorama<0.4.5,>=0.2.5
  Downloading colorama-0.4.4-py2.py3-none-any.whl (16 kB)
Collecting docutils<0.17,>=0.10
  Downloading docutils-0.16-py2.py3-none-any.whl (548 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 548.2/548.2 KB 50.9 MB/s eta 0:00:00
Collecting PyYAML<5.5,>=3.10
  Using cached PyYAML-5.4.1.tar.gz (175 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting rsa<4.8,>=3.1.2
  Downloading rsa-4.7.2-py3-none-any.whl (34 kB)
Collecting jmespath<2.0.0,>=0.7.1
  Downloading jmespath-1.0.1-py3-none-any.whl (20 kB)
Collecting urllib3<1.27,>=1.25.4
  Downloading urllib3-1.26.15-py2.py3-none-any.whl (140 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 140.9/140.9 KB 30.4 MB/s eta 0:00:00
Collecting python-dateutil<3.0.0,>=2.1
  Downloading python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 247.7/247.7 KB 44.6 MB/s eta 0:00:00
Collecting pyasn1>=0.1.3
  Downloading pyasn1-0.4.8-py2.py3-none-any.whl (77 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 77.1/77.1 KB 15.0 MB/s eta 0:00:00
Collecting six>=1.5
  Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)
Building wheels for collected packages: PyYAML
  Building wheel for PyYAML (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for PyYAML (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [574 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.linux-x86_64-3.11
      creating build/lib.linux-x86_64-3.11/yaml
      copying lib3/yaml/reader.py -> build/lib.linux-x86_64-3.11/yaml
      copying lib3/yaml/tokens.py -> build/lib.linux-x86_64-3.11/yaml
      copying lib3/yaml/emitter.py -> build/lib.linux-x86_64-3.11/yaml
      copying lib3/yaml/events.py -> build/lib.linux-x86_64-3.11/yaml
      copying lib3/yaml/nodes.py -> build/lib.linux-x86_64-3.11/yaml
      copying lib3/yaml/scanner.py -> build/lib.linux-x86_64-3.11/yaml
      copying lib3/yaml/constructor.py -> build/lib.linux-x86_64-3.11/yaml
      copying lib3/yaml/composer.py -> build/lib.linux-x86_64-3.11/yaml
      copying lib3/yaml/cyaml.py -> build/lib.linux-x86_64-3.11/yaml
      copying lib3/yaml/resolver.py -> build/lib.linux-x86_64-3.11/yaml
      copying lib3/yaml/parser.py -> build/lib.linux-x86_64-3.11/yaml
      copying lib3/yaml/serializer.py -> build/lib.linux-x86_64-3.11/yaml
      copying lib3/yaml/dumper.py -> build/lib.linux-x86_64-3.11/yaml
      copying lib3/yaml/loader.py -> build/lib.linux-x86_64-3.11/yaml
      copying lib3/yaml/__init__.py -> build/lib.linux-x86_64-3.11/yaml
      copying lib3/yaml/error.py -> build/lib.linux-x86_64-3.11/yaml
      copying lib3/yaml/representer.py -> build/lib.linux-x86_64-3.11/yaml
      creating build/lib.linux-x86_64-3.11/_yaml
      copying lib3/_yaml/__init__.py -> build/lib.linux-x86_64-3.11/_yaml
      running build_ext
      creating build/temp.linux-x86_64-3.11
      creating build/temp.linux-x86_64-3.11/yaml
      x86_64-linux-gnu-gcc -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -fPIC -I/usr/include/python3.11 -c yaml/_yaml.c -o build/temp.linux-x86_64-3.11/yaml/_yaml.o
      In file included from yaml/_yaml.c:744:
      yaml/_yaml.h:10: warning: "PyString_CheckExact" redefined
         10 | #define PyString_CheckExact PyBytes_CheckExact
            |
      yaml/_yaml.c:624: note: this is the location of the previous definition
        624 |   #define PyString_CheckExact          PyUnicode_CheckExact
            |
      yaml/_yaml.c: In function ‘__pyx_pf_4yaml_5_yaml_get_version_string’:
      yaml/_yaml.c:2062:17: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
       2062 |   __pyx_v_value = yaml_get_version_string();
            |                 ^
      yaml/_yaml.c: In function ‘__pyx_pf_4yaml_5_yaml_7CParser___init__’:
      yaml/_yaml.c:3592:52: warning: passing argument 2 of ‘yaml_parser_set_input’ from incompatible pointer type [-Wincompatible-pointer-types]
       3592 |     yaml_parser_set_input((&__pyx_v_self->parser), __pyx_f_4yaml_5_yaml_input_handler, ((void *)__pyx_v_self));
            |                                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
            |                                                    |
            |                                                    int (*)(void *, char *, size_t,  size_t *) {aka int (*)(void *, char *, long unsigned int,  long unsigned int *)}
      In file included from yaml/_yaml.h:2,
                       from yaml/_yaml.c:744:
      /usr/include/yaml.h:1370:30: note: expected ‘int (*)(void *, unsigned char *, size_t,  size_t *)’ {aka ‘int (*)(void *, unsigned char *, long unsigned int,  long unsigned int *)’} but argument is of type ‘int (*)(void *, char *, size_t,  size_t *)’ {aka ‘int (*)(void *, char *, long unsigned int,  long unsigned int *)’}
      ....
        note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for PyYAML
Failed to build PyYAML
ERROR: Could not build wheels for PyYAML, which is required to install pyproject.toml-based projects
```

### Possible Solution

Update PyYAML to 6.0+

### CLI version used

latest

### Environment details (OS name and version, etc.)

Ubuntu latest

**Comment by tim-finnigan** [comment]
Hi @kchoudhu thanks for reaching out. I tried to reproduce this but was unable to. (I could install awscli-1.27.96 on Python 3.11.2 on a Linux EC2 instance.) The issue could be that you're referencing a cached version of PyYAML. For me it installed `Downloading PyYAML-5.4.1-cp37-cp37m-manylinux1_x86_64.whl`. Have you tried clearing your package cache directory prior to installing the CLI?

**Comment by Ri0n** [comment]
not quite the same but pretty similar problem

```
$ docker run --rm -it --entrypoint="" public.ecr.aws/lambda/python:3.11 bash
bash-4.2# 
bash-4.2# pip install awscli
Collecting awscli
  Downloading awscli-1.29.3-py3-none-any.whl (4.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.1/4.1 MB 9.6 MB/s eta 0:00:00
Collecting botocore==1.31.3 (from awscli)
  Downloading botocore-1.31.3-py3-none-any.whl (11.0 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 11.0/11.0 MB 23.1 MB/s eta 0:00:00
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ReadTimeoutError("HTTPSConnectionPool(host='pypi.org', port=443): Read timed out. (read timeout=15)")': /simple/docutils/
Collecting docutils<0.17,>=0.10 (from awscli)
  Downloading docutils-0.16-py2.py3-none-any.whl (548 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 548.2/548.2 kB 28.4 MB/s eta 0:00:00
Requirement already satisfied: s3transfer<0.7.0,>=0.6.0 in /var/lang/lib/python3.11/site-packages (from awscli) (0.6.1)
Collecting PyYAML<5.5,>=3.10 (from awscli)
  Downloading PyYAML-5.4.1.tar.gz (175 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 175.1/175.1 kB 25.2 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [68 lines of output]
      /tmp/pip-build-env-6cfzy6dg/overlay/lib/python3.11/site-packages/setuptools/config/setupcfg.py:293: _DeprecatedConfig: Deprecated config in `setup.cfg`
      !!
      
              ********************************************************************************
              The license_file parameter is deprecated, use license_files instead.
      
              By 2023-Oct-30, you need to update your project and remove deprecated calls
              or your builds will no longer be supported.
      
              See https://setuptools.pypa.io/en/latest/userguide/declarative_config.html for details.
              ********************************************************************************
      
      !!
        parsed = self.parsers.get(option_name, lambda x: x)(value)
      running egg_info
      writing lib3/PyYAML.egg-info/PKG-INFO
      writing dependency_links to lib3/PyYAML.egg-info/dependency_links.txt
      writing top-level names to lib3/PyYAML.egg-info/top_level.txt
      Traceback (most recent call last):
        File "/var/lang/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/var/lang/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/var/lang/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-6cfzy6dg/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 341, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-6cfzy6dg/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 323, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-6cfzy6dg/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 338, in run_setup
          exec(code, locals())
        File "<string>", line 271, in <module>
        File "/tmp/pip-build-env-6cfzy6dg/overlay/lib/python3.11/site-packages/setuptools/__init__.py", line 107, in setup
          return distutils.core.setup(**attrs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-6cfzy6dg/overlay/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 185, in setup
          return run_commands(dist)
                 ^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-6cfzy6dg/overlay/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
          dist.run_commands()
        File "/tmp/pip-build-env-6cfzy6dg/overlay/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
          self.run_command(cmd)
        File "/tmp/pip-build-env-6cfzy6dg/overlay/lib/python3.11/site-packages/setuptools/dist.py", line 1234, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-6cfzy6dg/overlay/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/tmp/pip-build-env-6cfzy6dg/overlay/lib/python3.11/site-packages/setuptools/command/egg_info.py", line 314, in run
          self.find_sources()
        File "/tmp/pip-build-env-6cfzy6dg/overlay/lib/python3.11/site-packages/setuptools/command/egg_info.py", line 322, in find_sources
          mm.run()
        File "/tmp/pip-build-env-6cfzy6dg/overlay/lib/python3.11/site-packages/setuptools/command/egg_info.py", line 551, in run
          self.add_defaults()
        File "/tmp/pip-build-env-6cfzy6dg/overlay/lib/python3.11/site-packages/setuptools/command/egg_info.py", line 589, in add_defaults
          sdist.add_defaults(self)
        File "/tmp/pip-build-env-6cfzy6dg/overlay/lib/python3.11/site-packages/setuptools/command/sdist.py", line 104, in add_defaults
          super().add_defaults()
        File "/tmp/pip-build-env-6cfzy6dg/overlay/lib/python3.11/site-packages/setuptools/_distutils/command/sdist.py", line 251, in add_defaults
          self._add_defaults_ext()
        File "/tmp/pip-build-env-6cfzy6dg/overlay/lib/python3.11/site-packages/setuptools/_distutils/command/sdist.py", line 336, in _add_defaults_ext
          self.filelist.extend(build_ext.get_source_files())
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "<string>", line 201, in get_source_files
        File "/tmp/pip-build-env-6cfzy6dg/overlay/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 107, in __getattr__
          raise AttributeError(attr)
      AttributeError: cython_sources
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

[notice] A new release of pip is available: 23.1.2 -> 23.2
[notice] To update, run: pip install --upgrade pip
```

**Comment by tim-finnigan** [comment]
@Ri0n that issue is being tracked here: https://github.com/aws/aws-cli/issues/8036. Please see the latest update here: https://github.com/aws/aws-cli/issues/8036#issuecomment-1638544754.
